### Задача 1
Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

По условию мы знаем, что $n = 100$, $a = 17.5$, $\sigma^2 = 4$, необходимо проверить гипотезу: $H_0$: a = 17, тогда $H_1$: a > 17. Для проверки гипотезы используем выборочное среднее:
$$T(X) = \dfrac{1}{n} \displaystyle\sum_{i = 1}^n x_i,$$
где $X = (x_1, x_2, \dots, x_n)$.
Таким образом для выборки $T$ параметры будут следующие: $a = 17$, $\sigma = \sqrt{\dfrac{4}{n}}$.

Далее определим критическую область, для этого воспользуемся правилом двух сигм, но только  с одной стороны, поскольку наша критическая область правосторонняя, значит значения величины $T$ с вероятностью 0.975 попадают в интервал:

$$\Omega_\alpha = (17 + 2 \cdot \sqrt{\dfrac{4}{n}}, \infty)$$

Подставим значение $n = 100$ и получим критическую область:

$$(17.4, \infty)$$

In [1]:
import numpy as np

In [2]:
n = 100

samples = np.random.normal(17.5, 2, size=n)

print(samples)

[17.48820952 16.94533272 18.3842767  16.13724363 15.4319848  17.77762666
 14.28885965 18.79807504 17.69472988 17.54419332 18.10798482 16.38993622
 20.99884046 18.85359235 14.64847036 19.36003075 18.61774474 18.78473128
 17.23788958 16.54101171 18.70214963 12.06304944 13.56744614 18.98973925
 15.06079529 18.0514996  19.67147825 14.86001516 18.34184929 20.4636577
 19.00739219 17.69064333 17.31669555 17.47521446 16.21418157 17.7613244
 21.46231229 17.68418896 16.80724881 17.49429454 16.25516511 18.58846743
 17.69836156 15.22925741 14.95272085 16.0404794  20.97269823 17.07670818
 15.10169999 19.5206551  18.22719541 16.38651894 17.87415066 17.72976353
 18.652966   16.00561182 19.08325378 17.00863079 18.46521748 17.44857699
 20.59121257 12.42983996 15.89631847 17.09900489 15.13436896 19.34756714
 19.08883044 13.92122107 16.78621892 16.27254603 19.12018167 17.8240676
 19.47166876 16.85696453 15.57281727 18.53990928 17.4259296  16.82575544
 19.46740277 15.80058754 14.83694086 17.04725665 15.44

In [3]:
samples.mean()

17.328913771796124

Значение попало в критическую область, поэтому гипотеза $H_0$ отвергается в пользу гипотезы $H_1$

### Задача 2
Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания $a$ с надежностью 0.95, если выборочное среднее равно 80, а объём равен 256.

In [4]:
m = 80
n = 256
sigma = 16
samples = np.random.normal(m, sigma, size=n)
print(samples)

[ 60.15280921  75.19300638  53.57221809 104.2276589   64.05207706
  66.85403206 105.12114958  63.64020833  68.50172931 108.50787457
  82.09748898  74.16007854  97.05082752 103.24898669 116.35428627
 103.57897989  73.68252071 104.68708376  66.72597786  91.08471386
  66.20336426  57.41797718 125.57386063  71.87759601  91.94300928
  94.79503458  96.17128354 109.85249443  52.2894791   92.73104324
  71.06799728  95.38357894  77.03951574  68.92505759  89.22152032
  61.80187479  76.41566728  59.17123148  90.46446163  73.91967388
  75.03116434  81.40334827 104.28619958  89.74412043  62.78689508
  71.35068505  94.67351817 114.78705167  80.5251818   67.91107198
 102.79731263  78.84230574 100.58146008  76.11464753  72.81620706
  80.04950385  67.3823472   42.7694024   79.15154365  79.79252972
  74.95785095 102.87564813  58.91663876  91.68234426  65.37420734
 103.0122471   90.49627928  70.10196932  72.96383013 100.15073114
  76.68583868  86.12081495  84.61454029  77.28132079  90.57005252
  47.73238

Т.к. объем нашей выборки достаточно велик для получения квантиля воспользуемся функцией из библиотеки scipy

In [5]:
from scipy import stats

In [6]:
alpha = 0.05

In [7]:
t = stats.t.ppf(1 - alpha / 2, 255)

Теперь расчитаем доверительный интервал:

In [8]:
(m - t * sigma / np.sqrt(n), m + t * sigma / np.sqrt(n))

(78.0306894301508, 81.9693105698492)

### Задача 3
Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

202, 203, 199, 197, 195, 201, 200, 204, 194, 190

Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?

In [9]:
x = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
alpha = 0.01
n = 10
m = 200

Приверим гипотезу $H_0: m = 200$, для этого воспрользуемся __t-распределением__, поскольку нам неизвестни дисперсия.
Критическая область будет иметь вид:

$$\Omega_\alpha = \left( -\infty, - t_{1 - \alpha/2, \: n - 1} \right) \cup \left( t_{1 - \alpha/2, \: n - 1}, \infty \right).$$

Посчитаем выборочное среднее, несмещенную выборочную дисперсию и несмещенное среднеквадратичное отклонение.

In [10]:
X = np.mean(x)
s = np.std(x, ddof=1)
print(X, s)

198.5 4.453463071962462


In [11]:
t = stats.t.ppf(1 - alpha / 2, 9)
t

3.2498355440153697

$$\Omega_\alpha = \left( -\infty, -3.2498\right) \cup \left(3.2498, \infty \right).$$

In [12]:
T = (X - m) / (s * np.sqrt(n))
print(T)

-0.10651074037450896


In [13]:
T < t

True

In [14]:
T > -t

True

Так как $T$ не попадает в критическую область, гипотезу $H_0$ можно принять.